src/foam/core/XML.js

```diff
@@ -0,0 +1,466 @@
+          this.end('</' +  this.propertyName(p) + '>');
+        }
+      })
+    },
+
+    function outputPrimitive(v, p){
+      this.out('<').outputPropertyName(p).out('>');
+      this.output(p.toXML(v, this));
+      this.out('</').outputPropertyName(p).out('>');
+    },
+
+    function outputDate(o) {
+      if ( this.formatDatesAsNumbers ) {
+        this.out(o.valueOf());
+      } else {
+        this.out(JSON.stringify(o));
+      }
+    },
+
+    function outputFunction(o) {
+      if ( this.formatFunctionsAsStrings ) {
+        this.output(o.toString());
+      } else {
+        this.out(o.toString());
+      }
+    },
+
+    function outputObjectKeyValue_(key, value, first) {
+      if ( ! first ) this.out(',').nl().indent();
+      this.out(this.maybeEscapeKey(key), ':').output(value);
+    },
+
+    function outputObjectKeyValues_(o) {
+      var first = true;
+      for ( var key in o ) {
+        this.outputObjectKeyValue_(key, o[key], first);
+        first = false;
+      }
+    },
+
+    function outputSortedObjectKeyValues_(o) {
+      var key, keys = [];
+
+      for ( key in o ) keys.push(key);
+      keys.sort();
+
+      var first = true;
+      for ( var i = 0 ; i < keys.length; i++ ) {
+        key = keys[i];
+        this.outputObjectKeyValue_(key, o[key], first);
+        first = false;
+      }
+    },
+
+    {
+      name: 'output',
+      code: foam.mmethod({
+        Undefined: function(o) { this.out('null'); },
+        Null:      function(o) { this.out('null'); },
+        String:    function(o) { this.out(this.escape(o)); },
+        Number:    function(o) { this.out(o); },
+        Boolean:   function(o) { this.out(o); },
+        Date:      function(o) { this.outputDate(o); },
+        Function:  function(o) { this.outputFunction(o); },
+        AbstractEnum: function(o) { },
+        FObject:   function(o) {
+          if ( o.outputXML ) {
+            o.outputXML(this)
+            return;
+          }
+
+          var clsName = o.cls_.id;
+          this.start("<object class='" + clsName + "'>")
+          // Iterate through properties and output
+          var ps = o.cls_.getAxiomsByClass(foam.core.Property);
+          for ( var i = 0 ; i < ps.length ; i++ ) {
+            this.outputProperty_(o, ps[i]);
+          }
+          this.end('</object>');
+        },
+        Array: function(o) {
+          // Nested Objects and FObject Arrays Passed
+          for ( var i = 0 ; i < o.length ; i++ ) {
+            // Output 'value' tags for arrays containing non-FObject values
+            if ( !o[i].cls_) this.out("<value>");
+            this.output(o[i], this);
+            if ( !o[i].cls_ ) this.out("</value>");
+            if ( o.length - i != 1 ) this.nl().indent();
+          }
+        },
+        Object: function(o) {
+          if ( o.outputXML ) {
+            o.outputXML(this);
+          } else {
+            var oName = o.name;
+            this.start("<object name='" + oName + "'>");
```

Comment:
I don't think this (or the previous) escaping is needed since if the property contains a quote character, then you code isn't going to run anyway. Better to validate Property names to exclude quotes than add checks all over the code, especially when you would already be broken anyway.